docs/design/sql/20-match-jet-job-to-sql.md

```diff
@@ -0,0 +1,92 @@
+
+We can provide multiple approaches to solve given problem:
+
+### Reuse existing QueryId
+
+`QueryId` class appears to be a data structure for representing a query identifier. In fact, it is a 256-bit value,
+where upper 128 bit represents the query coordinator, and lower 128 bit represents the actual query id, which is
+generated by `UuidUtil.newUnsecureUUID()`.
+
+The basic idea of this approach is simple : we may take lower 64 bits of `QueryId` and use them as **Jet job**
+identifier during job creation in `PlanExecutor`:
+```java
+class PlanExecutor {
+    // ...
+    SqlResult execute(SelectPlan plan, QueryId queryId, List<Object> arguments, long timeout, SqlSecurityContext ssc) {
+        // ... 
+        long jobId = queryId.getLocalIdLow();
+        // ...
+        Job job = jet.newLightJob(jobId, plan.getDag(), jobConfig, ssc.subject());
+    }
+    // ...
+}
+```
+
+This way has one disadvantage, but with possible mitigation: Jet relies on a flake ID generator to create job IDs, 
+ensuring that the generated values are guaranteed to be unique across the entire cluster. However, when using 
+`UuidUtil.newUnsecureUUID()`, the same level of uniqueness cannot be guaranteed. Consequently, ID collisions 
+is just matter of time. Moreover, the client is always 'in enemy hands' and this fact also may cause same troubles.
+
+But we can easily overcome this issue by simply checking first if such job already exists in SQL job
+result registry, and only then start that job. If such job already exists - fail query with `SqlErrorCode.RESTARTABLE_ERROR`.
+
+The only thing that confuses us is that it might be a horrible developer experience, when user submits a query, and
+it fails due to internal job id collision. But, this behavior is also easily avoidable - SQL client has ability
+to set up an automatic query resubmission with `ClientSqlResubmissionMode`. Also, worth notice, that the probability
+of such collision is extremely low, so it should not be a problem for the usual user.
+
+### Return `jobId` to the client as a part of `SqlResult`
+
+This approach differs from the previous one in that it doesn't use `QueryId` at all. Instead, we are making an
+additional field in `SqlResult` (or `SqlClientResult`) class, which will contain `jobId` of the corresponding Jet job.
+
+This seemingly minor alteration, when examined closely, carries significant implications:
+we must now match Jet jobs with SQL queries during query execution, ideally even before the Jet job start its execution.
+The challenge with this approach is that presently (as of HZ 5.3), for small batch queries, as well as for all DML queries, 
+the `SqlResult` is not available until after the job has finished.  To address this predicament, we need
+to rework our query lifecycle to operate asynchronously: we'll return an empty `SqlResult` (or empty analogue
+of `SqlResult`) to the client and prompt the client to `fetch` the results. Worth noticing, it also solves a problem 
```

Comment:
it is easier to change member to return new kind of `SqlResult` than to update the way how it is handled on client side (seems very complicated, "optimised" and highly coupled)